vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher_test.go

```diff
@@ -38,6 +42,26 @@ var trueVar = true
 func simpleControllerFetcher() *controllerFetcher {
 	f := controllerFetcher{}
 	f.informersMap = make(map[wellKnownController]cache.SharedIndexInformer)
+	versioned := map[string][]metav1.APIResource{
+		"Foo": []metav1.APIResource{metav1.APIResource{Kind: "Foo", Name: "bah", Group: "foo"}},
+	}
+	fakeMapper := []*restmapper.APIGroupResources{
+		&restmapper.APIGroupResources{
+			Group: metav1.APIGroup{
+				Name:     "Foo",
+				Versions: []metav1.GroupVersionForDiscovery{metav1.GroupVersionForDiscovery{GroupVersion: "Foo", Version: "Foo"}},
+			},
+			VersionedResources: versioned,
+		},
+	}
+	mapper := restmapper.NewDiscoveryRESTMapper(fakeMapper)
+	f.mapper = mapper
+
+	clientSet := fake.NewSimpleClientset()
+	baseClient := clientSet.CoreV1().RESTClient()
+	resolver := scale.NewDiscoveryScaleKindResolver(clientSet.Discovery())
+	scaleNamespacer := scale.New(baseClient, mapper, dynamic.LegacyAPIPathResolverFunc, resolver)
+	f.scaleNamespacer = scaleNamespacer
```

Comment:
So it looks like RESTClient() of the fake ClientSet is nil (guess this is part of it being fake :) )

How about trying this instead:

```
scaleNamespacer := &scalefake.FakeScaleClient{}
f.scaleNamespacer = scaleNamespacer
```

This needs and import of 
scalefake "k8s.io/client-go/scale/fake" 
which is not in vendor, but if you can just copy it into vendor for now, I'll resolve the dependencies for you when we get this to work.